tofcamera_mesasr.orogen

```diff
@@ -12,6 +12,12 @@ using_library "vizkit3d"
 task_context "Task" do
    needs_configuration
 
+  property('interface', 'std::string', 'ETHERNET').
+	 doc ('Type of interface of the camera sensor. So far Swiss Ranger Cameras are available with USB or Ethernet interfaces for data connection. Possible values: USB, ETHERNET')
+
```

Comment:
Hi Matthias,

That's also an option yes.

The Task.cpp only checks for IP address if the ETHERNET interface is
selected and I tried to make the documentation clear enough for that but
ok, your solution is more robust to errors.
Regards,



2015-09-09 13:41 GMT+02:00 Matthias Goldhoorn <notifications@github.com>:

> In tofcamera_mesasr.orogen
> <https://github.com/rock-drivers/drivers-orogen-tofcamera_mesasr/pull/3#discussion_r39032152>
> :
>
> > @@ -12,6 +12,12 @@ using_library "vizkit3d"
> >  task_context "Task" do
> >     needs_configuration
> >
> > +  property('interface', 'std::string', 'ETHERNET').
> > +	 doc ('Type of interface of the camera sensor. So far Swiss Ranger Cameras are available with USB or Ethernet interfaces for data connection. Possible values: USB, ETHERNET')
> > +
>
> I recomment to use a URI based format like:
> serial://
> ip://
>
> and parse this within one 'port' member for deice access.
> Otherwise it could confuse on the interface.
>
> Otherwise use a enum for the type, and make sure ip is only set when
> ETHERNET is set, and vice-versa for the serial port.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rock-drivers/drivers-orogen-tofcamera_mesasr/pull/3/files#r39032152>
> .
>